### PR TITLE
Document naming conventions

### PR DESCRIPTION
**Pull request by sarjon**
Documents naming conventions discussed in https://github.com/PrestaShop/PrestaShop/issues/10597. 

It still needs proofreading and suggestions for improvements.

Todo:
- [x] Document services key naming
- [x] Grid naming

### CODE REVIEW

**Comment by matks** [comment]
Let's add this one 😄 
https://github.com/PrestaShop/PrestaShop/pull/10783/files#r240071710

**Comment by sarjon** [comment]
i've noticed one interesting thing, how do we structure templates with multiple partials (blocks)? 

should we document this convention, that suggest to split partials (block) by page? @matks wdyt?
```
- Blocks
    - View # all the blocks for main view.html.twig page
        - personal_information.html.twig
        - addresses.html.twig
        - carts.html.twig
    - Index # all the blocks for main index.html.twig page
        - required_fields.html.twig
- view.html.twig # main page
- create.html.twig # main page
- index.html.twig # mian page
```

File: src/content/1.7/development/naming-conventions/_index.md [hunk 1]
```diff
@@ -137,4 +139,32 @@ admin_customers_transform_guest_to_customer:
...
+  prestashop.core.payment.payment_option_form_decorator:
+    class: 'PrestaShop\PrestaShop\Core\Payment\PaymentOptionFormDecorator'
+```
+
+### Grid
+
+PrestaShop comes with a lot of Grids (Products, Customers, Orders & etc) and keeping consistency between them is very important, thats why it follows these naming conventions:
+
+- Grid id should be in lowercase and written in `snake_case`
+
```
Comment (mickaelandrieu) [inline src/content/1.7/development/naming-conventions/_index.md]: you could add an exemple (copy/pasted from your module).

Comment (mickaelandrieu) [reply src/content/1.7/development/naming-conventions/_index.md]: I'll add an exemple in Grid docs

**Comment by mickaelandrieu** [comment]
Thanks @sarjon, if something is missing here

You can create a new pull request 👍

**Pull request merged**